website/content/docs/plugins/containerized-plugins.mdx

```diff
@@ -0,0 +1,253 @@
+  registration process.
+
+## Running containerized plugins
+
+### Installation
+
+1. Install a container engine, for example:
+    * [Docker](https://docs.docker.com/engine/install/)
+    * Or [rootless Docker](https://docs.docker.com/engine/security/rootless/)
+
+2. [Install runsc](https://gvisor.dev/docs/user_guide/install/)
+
+3. Configure runsc to allow Unix sockets between the host and plugin, updating
+   the runsc install location if you didn't use the default:
+
+    a. Default Docker install:
+        ```bash
+        sudo tee /etc/docker/daemon.json <<EOF
+        {
+        "runtimes": {
+          "runsc": {
+            "path": "/usr/local/bin/runsc",
+              "runtimeArgs": [
+                "--host-uds=all"
+              ]
+            }
+          }
+        }
+        EOF
+        sudo systemctl reload docker
+        ```
+    
+    b. Rootless Docker install:
+        ```bash
+        mkdir -p ~/.config/docker
+        tee ~/.config/docker/daemon.json  <<EOF
+        {
+          "runtimes": {
+            "runsc": {
+              "path": "/usr/local/bin/runsc",
+              "runtimeArgs": [
+                "--host-uds=all"
+                "--ignore-cgroups"
+              ]
+            }
+          }
+        }
+        EOF
+        systemctl --user restart docker
+        ```
+
+### Building plugin container images
+
+To be compatible with running in a container, plugins must be built with at
+least v1.5.0 of [go-plugin](https://github.com/hashicorp/go-plugin). This can
+be pulled in directly via `go get github.com/hashicorp/go-plugin@latest`, or
+through the SDK from at least v0.10.0: `go get github.com/hashicorp/vault/sdk@latest`.
+
+Plugin container images should run a binary the same way it would be expected
+to run as a plain plugin process. For most plugins this simply means running
+the binary as the entrypoint. For example, for the KV plugin:
+
+1. Build the binary
+    ```bash
+    git clone https://github.com/hashicorp/vault-plugin-secrets-kv.git
+    cd vault-plugin-secrets-kv/
+    GOOS=linux go build -o kv cmd/vault-plugin-secrets-kv/main.go
+    ```
+
+1. Create the following `Dockerfile`:
+    ```Dockerfile
+    FROM gcr.io/distroless/static-debian12
+
+    COPY kv /bin/kv
+
+    ENTRYPOINT [ "/bin/kv" ]
+    ```
+
+1. Build the image:
+    ```bash
+    docker build -t hashicorp/vault-plugin-secrets-kv .
+    ```
+
+The image entrypoint, command, and environment can be further configured at
+registration time by using the `-command`, `-args`, and `-env` flags for
+`vault plugin register`.
+
+### Registration
+
+Once Vault has a suitable container runtime available, the remaining steps are
+very similar to registering and using other external plugins. The config setting
+for [`plugin_directory`](/vault/docs/configuration#plugin_directory) is not required
+for containerized plugins, but any plugin containers must already be available
+locally when mounted.
```

Comment:
```suggestion
Once you containerize the plugins, registering them with Vault is similar to
registering any other external plugins as long as the containers are available
locally when mounted.
```
Focus on what is required rather than what is not required since we're outlining steps the reader should take

Style correction: write in active voice,